assets/src/dashboard/components/popoverMenu/test/story-menu-generator.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import { noop } from '../../../../design-system';
+import {
+  STORY_CONTEXT_MENU_ITEMS,
+  STORY_CONTEXT_MENU_ACTIONS,
+} from '../../../constants';
+import { generateStoryMenu } from '../story-menu-generator';
+
+const TEST_MENU_ITEM_ACTIONS = Object.values(STORY_CONTEXT_MENU_ACTIONS).reduce(
+  (all, actionName) => {
+    all[actionName] = noop;
+
+    return all;
+  },
+  {
+    default: noop,
+  }
+);
+
+const TEST_STORY = {
+  previewLink: 'preview-link',
+  bottomTargetAction: 'bottom-target-action',
+};
+
+describe('generateStoryMenu', () => {
+  it('should generate menu items of the correct shape', () => {
+    const menuItems = generateStoryMenu({
+      menuItemActions: TEST_MENU_ITEM_ACTIONS,
+      menuItems: STORY_CONTEXT_MENU_ITEMS,
+      story: TEST_STORY,
+    });
+
+    expect(menuItems).toStrictEqual([
+      expect.objectContaining({
+        href: 'bottom-target-action',
+        label: 'Open in editor',
+        newTab: false,
+        onClick: expect.any(Function),
```

Comment:
The tests aren't testing that the correct functions were assigned to the correct list items. The `value` is used to associate the event handler with the list item, but the `value` is not returned on the final list item so it's tricky to identify which ones are which.

We could use the labels, but any update to the labels will break this test. I think that wouldn't be a good thing.

This is another indication that there must be a simpler way to create the menu items with the correct props and test them easily.

Reply:
moved to #6516